/* Copyright (C) 2014-2015 Alexander Shishenko <GamePad64@gmail.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
syntax = "proto3";
package librevault.protocol;

message Handshake {
	string user_agent = 1;
	bytes dir_hash = 2;
	bytes auth_token = 3;
}

message MetaList {
	message MetaRevision {
		bytes path_id = 1;
		int64 revision = 2;
	}
	repeated MetaRevision list = 1;
}

message MetaRequest {
	bytes path_id = 1;
}

message Meta {
	bytes meta = 1;
	bytes signature = 2;
}

message BlockRequest {
	bytes block_id = 1;
}

message Block {
	bytes block_id = 1;
	bytes block_content = 2;
}
